frontend/public/components/cluster-service-plan.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-undef */
+
+import * as React from 'react';
+import { ColHead, DetailsPage, List, ListHeader, ListPage, ResourceRow } from './factory';
+import { SectionHeading, detailsPage, navFactory, ResourceLink, ResourceSummary } from './utils';
+// eslint-disable-next-line no-unused-vars
+import { K8sResourceKind, K8sResourceKindReference } from '../module/k8s';
+import { viewYamlComponent } from './utils/vertnav';
+
+
+const ClusterServicePlanReference: K8sResourceKindReference = 'ClusterServicePlan';
+
+const ClusterServicePlanHeader: React.SFC<ClusterServicePlanHeaderProps> = props => <ListHeader>
+  <ColHead {...props} className="col-xs-4" sortField="metadata.name">Name</ColHead>
+  <ColHead {...props} className="col-xs-4" sortField="spec.externalName">External Name</ColHead>
+  <ColHead {...props} className="col-xs-4" sortField="spec.clusterServiceBrokerName">Broker</ColHead>
+</ListHeader>;
+
+const ClusterServicePlanListRow: React.SFC<ClusterServicePlanRowProps> = ({obj: servicePlan}) => <ResourceRow obj={servicePlan}>
+  <div className="col-xs-4 co-resource-link-wrapper">
```

Comment:
`co-resource-link-wrapper` -- I don't think this is needed unless there is a cog. @rhamilto can you confirm?